### PR TITLE
Add app illustration

### PR DESCRIPTION
**Pull request by skonto**
- Adds Cloudflow app illustration
- Updates readme 
-  Fixes dead links

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Cloudflow
+<p>
+<img src="./docs/images/logo.svg" width="300">
```
Comment (skonto) [inline README.md]: maybe there is better way than hardcoded values, but it seems to work (needs check).

File: README.md [hunk 2]
```diff
@@ -18,40 +20,48 @@ The different parts of Cloudflow work in unison to dramatically accelerate your
...
+
+<i>Basic components of a Cloudflow Application</i>
+</p>
 
 As we discussed above, Cloudflow allows developers to quickly build and deploy distributed stream processing applications by breaking such applications into smaller stream processing units called `Streamlets`.
 Each `Streamlet` represents an independent stream processing component that implements a self-contained stage of the application logic.
 `Streamlets` let you break down your application into logical pieces that communicate with each other in a streaming fashion to accomplish an end to end goal.
 
-`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology.
+`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology. The topology formed defines a DAG.
```
Comment (RayRoestenburg) [inline README.md]: Why do you want to add this? (It's also not exactly true, BTW, we do not stop anyone from creating a cycle)

Comment (RayRoestenburg) [reply README.md]: I would prefer to leave this addition out.

File: README.md [hunk 3]
```diff
@@ -18,40 +20,48 @@ The different parts of Cloudflow work in unison to dramatically accelerate your
...
 
-`Streamlets` expose one or more inlets and outlets that represent the data consumed and produced by the `Streamlet`, respectively. 
-Inlets and outlets are schema-driven, ensuring that data flows are always consistent and that connections between `streamlets` are compatible. 
+`Streamlets` expose one or more inlets and outlets that represent the data consumed and produced by the `Streamlet`, respectively.
+Inlets and outlets are schema-driven, ensuring that data flows are always consistent and that connections between `streamlets` are compatible.
+In the diagram above *Streamlet 1* has one outlet which feeds data to *Streamlet 2* inlet. *Streamlet 1* is a component that generates data or could get its data from an external system eg. via an http request.
+Then *Streamlet 2* outlet feeds its data output to *Streamlet 3* inlet. *Streamlet 2* in this application does the actual data processing.
+*Streamlet 3* then may store its data to some external system or just print to console.
```
Comment (RayRoestenburg) [inline README.md]:
```suggestion
*Streamlet 3* then may store its data to some external system.
```

Comment (RayRoestenburg) [reply README.md]: (just printing is not that useful)

File: README.md [hunk 4]
```diff
@@ -18,40 +20,48 @@ The different parts of Cloudflow work in unison to dramatically accelerate your
...
 A rapidly growing percentage of that data is now arriving in the form of data streams.
-To extract value from that data as soon as it arrives, those streams require near-realtime processing. 
+To extract value from that data as soon as it arrives, those streams require near-realtime processing.
 We use the term "Fast Data" to describe applications and systems that deal with such requirements.
 
 The Fast Data landscape has been rapidly evolving, with tools like Spark, Flink, and Kafka Streams emerging from the world of large-scale data processing while projects like Reactive Streams and Akka Streams have emerged from the world of application development and high-performance networking.
 
-The demand for availability, scalability, and resilience is forcing Fast Data architectures to become more like microservice architectures. 
-Conversely, successful organizations building microservices find their data needs grow with their organization while their data sources are becoming more stream-like and more real-time. 
+The demand for availability, scalability, and resilience is forcing Fast Data architectures to become more like microservice architectures.
```
Comment (RayRoestenburg) [inline README.md]:
```suggestion
The demand for availability, scalability, and resilience is forcing fast data architectures to become more like microservice architectures.
```

File: README.md (hunk 3 above)
Comment (skonto) [reply README.md]: it is for the dev lifecycle isn't?

File: README.md (hunk 2 above)
Comment (skonto) [reply README.md]: That is the correct definition of the topology, if you have one I would like to add it here.

File: README.md (hunk 4 above)
Comment (skonto) [reply README.md]: This was already there but I can change it.

File: README.md (hunk 3 above)
Comment (RayRoestenburg) [reply README.md]: you can print anywhere, you don't have to create a streamlet just for printing, is what I mean.

File: README.md (hunk 2 above)
Comment (RayRoestenburg) [inline README.md]:
```suggestion
`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology.
```

**Comment by skonto** [comment]
I updated the PR.

**Comment by skonto** [comment]
Ok guys I will merge thanks!

**Pull request merged**